whois-api/src/main/java/net/ripe/db/whois/api/rdap/WhoisRdapService.java

```diff
@@ -140,10 +140,10 @@ public Response lookup(@Context final HttpServletRequest request,
                         lookupObject(request, whoisObjectTypes, key);
             }
             case NAMESERVER: {
-                throw new ServerErrorException("Nameserver not supported", Response.Status.NOT_IMPLEMENTED);
+                throw new ServerErrorException("501 Not Implemented", Response.Status.NOT_IMPLEMENTED, new Throwable("Nameserver not supported"));
             }
```

Comment:
Feel free to ignore :

May be you can create a custom RdapException and have status code, errorTiltle and errorMessage as arguments. 
Its more readable and easier to throw new excpetion later on which requires detailed error message and also we can get rid of lot of if conditions in rdapExceptionMapper

Reply:
But remember not to remove QueryException , JsonProcessingException and others from RdapExcepionMapper, I do not think we care caching them in service layer.  Bu wherever you need detailed message ha would be way simpler

Reply:
I added RdapException with title, code and description and I am using it when we return an exception and doing the mapping. However, as you specified I am leaving the previous errors in the handler just in case we are not caching it